@xen-orchestra/fs/src/_mount.js

```diff
@@ -75,9 +75,17 @@ export default class MountHandler extends LocalHandler {
 
     try {
       const { type, device, options, env } = this._params
+
+      // Linux mount uses a different syntax than *BSD implementations.
+      // Use os.type to find out, which syntax to use.
+      const uname = ostype()
+      const mount_args = uname.includes('Linux') ?
+        ['-t', type, device, realPath, '-o', options] : // Linux
+        ['-o', options, '-t', type, device, realPath] 	// *BSD,...
```

Comment:
As far as I understand, it appears BSD simply requires flags to be put before other arguments, it should also with Linux, no?

Reply:
Thank you for the fast reply.

According to the  FreeBSD manpage for mount the synopsis is 
`mount [-dfpruvw] [-o options] [-t [no]type[,type ...]] special node`
special translates to device and node to realPath.

The Linux manpage shows the following synopsis
`mount [-fnrsvw] [-t fstype] [-o options] device dir`

I just did another test on FreeBSD 12 and Debian 10:
FreeBSD insists on the ordering *-o options* then *-t type*,
Debian does accept both.


This means we should be able to change the arguments to 
`['-o', options, '-t', type, device, realPath]` for both Linux and BSD?

Reply:
Yes, let's keep as simple as possible, please add a small comment explaining that this order is necessary for BSD :slightly_smiling_face: